### PR TITLE
Squeeze y before checking its shape

### PR DESCRIPTION
**Pull request by paweller**
Make sure that y is not of shape (num_samples, 1). np.squeeze will remove all axis with dimension 1.

### CODE REVIEW

**Comment by soroushj** [comment]
Thank you for the PR, @paweller.

I think we should allow a shape of `(n, 1)` for `y`. Obviously, `keras_balanced_batch_generator.make_generator` is not useful on such a dataset, but it works as expected. `(n, 1)` is generally a valid shape for `y`--it's a one-class dataset, there's nothing special about it. As an example, if you set `num_classes = 1` in [`example_multiclass`](https://github.com/soroushj/keras-balanced-batch-generator/blob/v0.0.1/example_multiclass.py#L9), it runs normally (and, of course, you'll get an accuracy of 100%).

Also, checking `len(np.squeeze(y).shape) == 2` introduces another problem: Now `(n, m, 1)`, `(n, m, 1, 1)`, etc. are considered valid shapes.

So I think no change is necessary. Checking `len(y.shape) == 2` works just fine.

If you still think there's a good reason for this change, I'm happy to discuss it. Otherwise, we can close this PR.

Again, thanks for your contribution.

**Comment by paweller** [comment]
Thanks for your response.

Regarding your first paragraph, I am totally with you. In case of a one-class setup `(n, 1)` would be valid. Is there any real-world use case to this setup though?

Regarding your second paragraph: Good point. Did unfortunately not think about that.

Regarding the rest: The reason behind my thoughts to use `np.squeeze()` has been the following: Suppose, a multi-class vector is read in as `y.shape == (n_classes, 1)` or as `y.shape == (n_classes,)` followed by a `np.expand_dims(y, -1)` somewhere before feeding it to the batch generator. `len(y.shape)` gives `2` as a result. Thus, `if len(y.shape) != 2:` will not raise a `ValueError` even though `y` is not a binary class matrix.

To make everything bullet-proof the checking process would probably have to be a little bit more involved. It could be solved by adding the `num_classes` parameter to the `make_generator()` function's arguments and using it to verify that shape `(n, 1)` actually is a valid shape. Something like this maybe (?):

```python
if len(y.shape) != 2:
    raise ValueError(
        'Arg `y` must have a shape of (num_samples, num_classes). ' +
        'You can use `keras.utils.to_categorical` to convert a class vector ' +
        'to a binary class matrix.'
    )
if num_classes != y[1].shape:
    raise ValueError('y[1].shape == {} must equal num_classes == {}'.format(y[1].shape, num_classes))
```

**Comment by soroushj** [comment]
The [guide](https://github.com/soroushj/keras-balanced-batch-generator/blob/v0.0.1/README.md#api) explains how to convert a class vector into a binary class matrix. I see your point, but I don't really want to turn this into a validation library. Even what you suggested is not actually enough to check that `y` is a valid binary class matrix--for that, you have to check every single element. The important thing is that, for any `y` of shape `(n, m)`, the behavior is clear and consistent (we take the argmax), and no runtime error occurs. On the other hand, `make_generator` *should* catch and report errors that, if not caught, will lead to a cryptic runtime error (see #3 as an instance).
